src/models/timeline/iterator.rs

```diff
@@ -0,0 +1,193 @@
+use super::Timeline;
+use crate::models::slot::{iterator::SlotIterator, Slot};
+use chrono::Duration;
+
+/*
+
+TimelineIterator goals:
+- ability to cros over timeline slots like SlotIterator through interval duration
+    - cross over timeline by 1 day duration
+- Get count of days, hours, minutes, etc for a timeline
+
+Ideas:
+- Ability to move to a specific time in the timeline.
+- Ability to perform some action in a timeline without
+forcing to split slots into hours, or similar.
+-
+*/
+
+/// Iterator for a `Timeline` and provide some functionalities like count per duration interval
+#[derive(Debug, Clone)]
+pub struct TimelineIterator {
+    timeline: Timeline,
+    pointer: usize,
+    /// Duration interval for pointer to corss over timeline timelines
+    interval: Duration,
+}
+
+impl TimelineIterator {
+    /// Initialize new TimelineIterator with default interval duration to 1 day
+    pub fn initialize(timeline: Timeline) -> TimelineIterator {
+        if let Some(_pionter_start) = timeline.slots.first() {
+            TimelineIterator {
+                timeline: timeline.clone(),
+                pointer: 0,
+                interval: Duration::days(1),
+            }
+        } else {
+            panic!("Timeline slots are empty")
+        }
+    }
+
+    /// Create new TimelineIterator with custom interval duration
+    pub fn new(timeline: Timeline, interval_duration: Duration) -> TimelineIterator {
+        if let Some(_pionter_start) = timeline.slots.first() {
+            TimelineIterator {
+                timeline: timeline.clone(),
+                pointer: 0,
+                interval: interval_duration,
+            }
+        } else {
+            panic!("Timeline slots are empty")
+        }
+    }
+
+    /// Get count between start and end for all slots in
+    /// timeline based on Duration interval
+    pub fn interval_count(&self) -> usize {
```

Comment:
Why is this functionality needed?  
I don't see it being called anywhere?

Reply:
Actually I found many functions which not called yet, so I'm going to remove them gradually when no need for them.